pkg/services/grpcserver/interceptors/logging.go

```diff
@@ -0,0 +1,22 @@
+package interceptors
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"google.golang.org/grpc"
+)
+
+func LoggingUnaryInterceptor(logger log.Logger) grpc.UnaryServerInterceptor {
+	return func(
+		ctx context.Context,
+		req any,
+		info *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (resp any, err error) {
+		logger = logger.FromContext(ctx)
+		resp, err = handler(ctx, req)
+		logger.Info("gRPC call received", "method", info.FullMethod, "req", req, "resp", resp, "err", err)
```

Comment:
a few thoughts:
* should `logger.Error` be used if `err != nil`?
* it looks like the http logging middleware is [controlled by a config option](https://github.com/grafana/grafana/blob/83438bd2318ceab0fcf1684ece3b9a6ab017d5f0/pkg/middleware/loggermw/logger.go#L98), which is disabled by default. would it make sense to do something similar here?
* could this possibly expose sensitive information by logging the full request and response? not sure if it is a relevant example for the gRPC server, but one example i'm thinking of is the decrypted data source config sent in requests to plugins

Reply:
I like the idea of having a config option for it and having it off by default. I think, at least for now, it would solve your third point. While were rolling this out for playlists, having everything logged would be helpful and harmless afaik. Once other services are brought into unified storage, we'd have to revisit and make our logging a little less verbose.

Reply:
that makes sense to me as long as it's clear to the user about what is currently being logged when it is enabled